compiler/testData/codegen/boxInline/smap/crossroutines.kt

```diff
@@ -1,4 +1,4 @@
-// This test depends on line numbers
+// IGNORE_BACKEND_MULTI_MODULE: JVM_IR
```

Comment:
At lines 128 and 142, source name (I guess?) became `_2Kt`, instead of `_2Kt$box$1`. @pyos can you spot what could go wrong here?

Reply:
It's because you're using the wrong class when creating a fake `ClassCodegen`.

```kotlin
fun f() {
    someNonInlineFunction {
        someInlineFunction {
            println("Hello, World!")
        }
    }
}
```

is transformed into

```kotlin
class AKt {
    fun f() {
        someNonInlineFunction(`AKt$f$1`())
    }
    fun `f$lambda-1`() {
        println("Hello, World!")
    }
}

class `AKt$f$1` : ... {
    override fun invoke() {
        someInlineFunction(AKt::`f$lambda-1`)
    }
    // + other stuff
}
```
Note that when inlining `someInlineFunction`, `codegen.classCodegen.irClass` is `AKt$f$1`, but the parent class of `f$lambda-1` is `AKt`.

In any case, there's still the matter of inner classes, so this approach is not fully correct either.

Reply:
Thanks for the pointer. Using the right parent IrClass made this test see correct SMAP again.

I didn't close your other comment above, so let's move on to there.